libs/langchain-core/src/callbacks/tests/callbacks.test.ts

```diff
@@ -242,6 +242,10 @@ test("CallbackManager Chat Message Handling", async () => {
       await llmCb.handleLLMEnd({ generations: [] });
     })
   );
+
+  // In case background mode is on while running this test
+  await awaitAllCallbacks();
```

Comment:
This has been a race condition that was unnoticed before. `p-queue` v6 had synchronous task chaining:
```ts
_next() {
    this._pendingCount--;
    this._tryToStartAnother();  // Synchronously starts next task
    this.emit('next');
}

// In _tryToStartAnother():
job();  // Calls the job synchronously
```
while v9 uses microtask-deferred task chaining:
```ts
// In the add() method's finally block:
queueMicrotask(() => {
    this.#next();  // Deferred to next microtask!
});
```